iceoryx2-bb/lock-free/src/mpmc/container.rs

```diff
@@ -452,6 +492,34 @@ impl<T: Copy + Debug, const CAPACITY: usize> FixedSizeContainer<T, CAPACITY> {
         unsafe { self.state.add(value) }
     }
 
+    /// Adds a new element to the [`FixedSizeContainer`]. If there is no more space available it returns
+    /// [`None`], otherwise [`Some`] containing the the index value to the underlying element.
+    ///
+    /// ```
+    /// use iceoryx2_bb_lock_free::mpmc::container::*;
+    ///
+    /// const CAPACITY: usize = 139;
+    /// let container = FixedSizeContainer::<u128, CAPACITY>::new();
+    ///
+    /// match unsafe { container.add_raw(1234567) } {
+    ///     Some(index) => {
+    ///         println!("added at index {}", index);
+    ///         unsafe { container.remove_raw_index(index) };
+    ///     },
+    ///     None => println!("container is full"),
+    /// };
+    ///
+    /// ```
+    ///
+    /// # Safety
+    ///
+    ///  * Use [`FixedSizeContainer::remove_raw_index()`] to release the acquired index again. Otherwise,
+    ///     the element will leak.
+    ///
+    pub unsafe fn add_raw(&self, value: T) -> Option<u32> {
+        self.state.add_raw(value)
```

Comment:
Since there is also a `ContainerState` struct in this PR, I had to look it up and was surprised that `state` is actually a `Container` and not `ContainerState`. Currently I don't have a good suggestion but if you don't mind I can make some proposals and we might create a terminology/jargon file to collect the names for these patterns.

If I don't misunderstand the meaning of the current `state` member, it basically contains the business logic for the `FixedSizeContainer`, right?

Also nothing to block this PR.

Reply:
I renamed the `state` member to `container`, since it contains the runtime fixed size container.